### PR TITLE
feat: WinUI InfoBadge

### LINKED ISSUE #6952: Support for WinUI `InfoBadge`
## What would you like to be added:

Support for `InfoBadge` control.

## Why is this needed:

WinUI parity

## For which Platform:

All

### PR DESCRIPTION
**Pull request by DZetko**
GitHub Issue (If applicable): #6952

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

InfoBadge doesn't work.

## What is the new behavior?

InfoBadge works.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/InfoBadge/InfoBadgeTemplateSettings.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		}
+
+		public IconElement IconElement
+		{
+			get => (IconElement)GetValue(IconElementProperty);
+			set => SetValue(IconElementProperty, value);
+		}
+
+		public static readonly DependencyProperty IconElementProperty =
+			DependencyProperty.Register(nameof(IconElement), typeof(IconElement), typeof(InfoBadgeTemplateSettings), new FrameworkPropertyMetadata(null));
```
Comment (carldebilly) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/InfoBadge/InfoBadgeTemplateSettings.cs:22]:
```suggestion
		public static readonly DependencyProperty IconElementProperty { get; } =
			DependencyProperty.Register(nameof(IconElement), typeof(IconElement), typeof(InfoBadgeTemplateSettings), new FrameworkPropertyMetadata(null));

```
In Uno sources, we're following the WinUI/WinRT way to declare DependencyProperties...

File: src/Uno.UI/Microsoft/UI/Xaml/Controls/InfoBadge/InfoBadge.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			{
+				VisualStateManager.GoToState(thisAsControl, "Value", true);
+			}
+			else if (IconSource is { } iconSource)
+		    {
+				TemplateSettings.IconElement = iconSource.CreateIconElement();
+				if (iconSource is FontIconSource)
+		        {
+					VisualStateManager.GoToState(thisAsControl, "FontIcon", true);
+				}
```
Comment (Youssef1313) [inline src/Uno.UI/Microsoft/UI/Xaml/Controls/InfoBadge/InfoBadge.cs:70]: Indentation isn't correct here

**Comment by jeromelaban** [comment]
Nice! Were there tests for this control ?

**Comment by MartinZikmund** [comment]
@jeromelaban There are tests, but I would still merge the control for 3.10, as it will be a nice addition to changelog :-) .

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**